chain/txValidator.go

```diff
@@ -682,13 +754,26 @@ func (bvs *BlockValidationState) CleanSubmittedTransactions(txns []*transaction.
 			subscribePayload := payload.(*pb.Subscribe)
 			topic := subscribePayload.Topic
 			bucket := subscribePayload.Bucket
-			key := subscription{topic, bucket, BytesToHexString(subscribePayload.Subscriber), subscribePayload.Identifier}
-			delete(bvs.subscriptions, key)
+			key := subscription{topic, bucket, string(subscribePayload.Subscriber), subscribePayload.Identifier}
+			if newSubscription, ok := bvs.subscriptions[key]; ok {
+				delete(bvs.subscriptions, key)
 
-			bvs.subscriptionCount[topic]--
+				if newSubscription {
+					bvs.subscriptionCount[topic]--
 
-			if bvs.subscriptionCount[topic] == 0 {
-				delete(bvs.subscriptionCount, topic)
+					if bvs.subscriptionCount[topic] == 0 {
+						delete(bvs.subscriptionCount, topic)
+					}
+				}
+			}
+		case pb.UNSUBSCRIBE_TYPE:
+			unsubscribePayload := payload.(*pb.Unsubscribe)
+			topic := unsubscribePayload.Topic
+			key := subscription{topic, 0, string(unsubscribePayload.Subscriber), unsubscribePayload.Identifier}
```

Comment:
Do you think it's a good idea to separate bvs.subscriptions and bvs.unsubscriptions ? I kind of feel that when someone send a sub and unsub txn at the same time, it's very easy to have bugs if we mix them up (but it's just my feeling)

Reply:
yeah, it's easy to have bugs, so it doesn't allow sending sub and unsub at the same time